src/avx512bw/avx512bw-from-utf8.inl.cpp

```diff
@@ -0,0 +1,462 @@
+/*
+validate_utf8_structure only checks if leading bytes
+are follewed by proper number of continuation bytes
+and then there's another leading byte.
+
+We don't handle end-of-stream here.
+*/
+bool validate_utf8_structure(__m512i input) {
+    /* 1. bitmask for various character byte classes.
+
+        leading: 111010011100011010001
+                 abccdddefgggghiijjjjk   a-k -- 11 UTF8- chars characters
+
+        ascii:   110000011000010000001
+                 ab     ef    h      k
+
+        2 bytes: 001000000000001000000
+                   cc          ii
+
+        3 bytes: 000010000000000000000
+                     ddd              
+
+        4 bytes: 000000000100010010000
+                          gggg   jjjj
+    */
+    uint64_t leading;
+    uint64_t ascii;
+    uint64_t _2bytes;
+    uint64_t _3bytes;
+    uint64_t _4bytes;
+
+    // we can valide 60 - 4 leading bytes
+    constexpr uint64_t mask  = 0x0ffffffffffffffflu;
+
+    {
+        const __m512i t0 = _mm512_and_si512(input, v_c0);
+        leading = _mm512_cmpneq_epu8_mask(t0, v_80);
+    }
+    {
+        ascii = _mm512_testn_epi8_mask(input, v_80) & mask;
```

Comment:
Possibly `&` could be replaced with _kand_mask64 so that it is more explicit.

Reply:
Later on, there are a lot of bit operations on these masks. I deliberately used `uint64_t` rather kmasks.

Reply:
Yes, I'm aware that converting masks back and forth is slow. I'll try to rewrite it to use the mask and then compare the compiler's output.

Reply:
Thank you, Daniel. It turned out that the resulting assembly is significantly shorter. (BTW, I forgot that kmask ops are three-argument ones.)